### ISSUE TITLE
findProgram improvements

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#1650]
From https://github.com/Macaulay2/M2/commit/30a21e5cb412be174940fd87f9b3cc8b431d2651#r44778844:

> Yes, it sounded familiar. I have no preference, and it's your file, so feel free to change it. But, umm, shouldn't findProgram filter off white space anyway? And also it could give a different error if the version number is not a version number. For example, if it's "-n 1.0". Also, it would be nice if instead of "could not find 0", it said something like "found '0' but the version number is '-n 1.0', which is less than '1.0'".

**Comment by d-torrance** [comment]
Summarizing Dan's comments, we would like `findProgram`'s `MinimumVersion` option to:

- [ ] strip off whitespace (then vanilla `echo` will work for the unit test)
- [ ] complain if the output doesn't look like a version number
- [ ] give a more verbose message when a version less than the minimum is found

**Comment by d-torrance** [comment]
@DanGrayson - Your third comment is already implemented if `Verbose` is set to `true`:

```m2
i9 : findProgram("gfan", "gfan --help", Verbose => true,
     MinimumVersion => ("1.0",
     "gfan _version | head -2 | tail -1 | sed 's/gfan//' | tr -d '\n'"))
checking for gfan in /usr/lib/x86_64-linux-gnu/Macaulay2/bin/...
    not found
checking for gfan in /home/profzoom/.local/bin/...
    not found
checking for gfan in /usr/local/sbin/...
    not found
checking for gfan in /usr/local/bin/...
    not found
checking for gfan in /usr/sbin/...
    not found
checking for gfan in /usr/bin/...
   found, but version 0.6.2 < 1.0
checking for gfan in /sbin/...
    not found
checking for gfan in /bin/...
   found, but version 0.6.2 < 1.0
checking for gfan in /usr/games/...
    not found
checking for gfan in /usr/local/games/...
    not found
checking for gfan in /snap/bin/...
    not found
stdio:11:1:(3): error: could not find gfan
```
Is that sufficient, or would it be good if the final error message also included the version information?

**Comment by DanGrayson** [comment]
The error message "foo not found" is not quite right, so it will put the user on the wrong track (if the program is found but the version number is bad).

**Comment by d-torrance** [comment]
> The error message "foo not found" is not quite right, so it will put the user on the wrong track (if the program is found but the version number is bad).

That makes sense!

**Comment by d-torrance** [comment]
This was addressed in #1665.  Closing
